types/node/index.d.ts

```diff
@@ -2342,7 +2342,19 @@ declare module "url" {
         query?: string | null | ParsedUrlQuery;
     }
 
-    export function parse(urlStr: string, parseQueryString?: boolean, slashesDenoteHost?: boolean): Url;
+    export interface UrlParsedQuery extends Url {
+        query: ParsedUrlQuery;
+    }
+
+    export interface UrlStringQuery extends Url {
```

Comment:
Perhaps rename these types to `UrlWith{StringQuery,ParsedQuery}` for readability?